### PR TITLE
LAU-496 Fix null pointer exception for lau case refs.

### PR DESCRIPTION
**Pull request by MartinYSpasov**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-496

### Change description ###
Fix null pointer exception for lau case refs.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by davejones74 — APPROVED** [review]

**Review by paulridout — COMMENTED** [review]
Might be an idea to add a test class for SearchLog that explicitly tests the behaviour of setCaseRefs().  The existing CaseSearchPostRequestTest.jsonContructionTestWithEmptyCaseRefs() covers the scenario where the list of case refs is null, but it uses the constructor to set up the class rather than the setCaseRefs() method.  Adding a test class would improve coverage and prove that the change prevents a null pointer exception.

**Review by paulridout — APPROVED** [review]
Changes look OK.

File: src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        assertThat(searchLog.getUserId()).isEqualTo(userId);
+        assertThat(searchLog.getTimestamp()).isEqualTo(timestamp.format(ISO_INSTANT));
+        assertThat(searchLog.getCaseRefs().size()).isEqualTo(3);
+        assertThat(searchLog.getCaseRefs().get(0)).isEqualTo("123");
+        assertThat(searchLog.getCaseRefs().get(1)).isEqualTo("567");
+        assertThat(searchLog.getCaseRefs().get(2)).isEqualTo("789");
+    }
+
+    @Test
+    void shouldNotThrowNpeWhenCaseRefIsNull() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java:52]: We need to add another test case to pass null value for caseRefs through SearchLog constructor. (as highlighted by Paul too)
Ex:  final SearchLog searchLog = new SearchLog(userId, null, timestamp);

File: src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        final SearchLog searchLog = new SearchLog();
+        final String caseRefs = null;
+        searchLog.setCaseRefs(caseRefs);
+
+        assertThat(searchLog.getCaseRefs()).isNull();
+    }
+
+    @Test
+    void shouldTrimCaseRefs() {
+        final SearchLog searchLog = new SearchLog();
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java:62]: We can add another test case to check the trim through SearchLog constructor.
Ex: caseRefs = asList(" 123 "," 567 "," 789  ");

File: src/main/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLog.java [hunk 3]
```diff
@@ -19,8 +20,6 @@ public class SearchLog implements Serializable {
 
     public static final long serialVersionUID = 432973322;
 
-    private static final String ANY_SPACE_MATCH = "\\s*";
-
     private String userId;
     private List<String> caseRefs;
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLog.java:24]: Should we need to initiate with empty list to avoid NPE for the caller?

**Review by rajaram-nagarajan — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java (hunk 1 above)
Comment (MartinYSpasov) [reply src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java:53]: SearchLog constructor was removed since it was used for testing purposes only.

File: src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java (hunk 2 above)
Comment (MartinYSpasov) [reply src/test/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLogTest.java:63]: SearchLog constructor was removed since it was used for testing purposes only.

File: src/main/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLog.java (hunk 3 above)
Comment (MartinYSpasov) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/lau/SearchLog.java:24]: No need, caseRefs can be null.

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM
